app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java

```diff
@@ -207,7 +222,18 @@ private void loadFragment(int navItemId) {
                 fragment = EventDashboardFragment.newInstance(eventId);
                 break;
         }
-        fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).commit();
+
+        if(fragment_count>1){
+            fragmentManager.popBackStack("root_fragment",0);
+            fragment_count--;
+        }
```

Comment:
We only need a boolean field because we won't go beyond 1 level

Reply:
Yes, I totally agree with you, we won't go beyond 1 level. But here suppose we go from fragments A(dashboard) to B to C then pressing back would lead to A. But in the process firstly A is added followed by B and now as we move to C, B is removed so, that backstack is AC . So, we need to keep a counter for this. How could I use boolean field I am not getting it?

Reply:
a->b->c should be reduced to a->c because we want to keep the navigation from event dashboard flat. So that no matter how many navigation a person performs, a single back click makes him come down to event dashboard

Reply:
Yes, so for that we need to count the fragments in the stack and max 2 fragments can be in stack one is dashboard and other one is the other fragments. Thats why I have used the counter.

Reply:
boolean field can be used if we needed to take care of one fragment only but here we need to take care of two fragments. The app is designed such that when it is created a fragment(dashboard or eventlist) is laid over the MainActivity so we need to add this fragment to the backstack and now if we select another fragment from nav bar then we need to add this one to the backstack and so total elements in backstack equals 2. So in this case we can't use boolean field.

Reply:
Yeah so at the time of pushing, you check the size of stack to see if the limit is reached and if it is, set the boolean value. And check that boolean value here. A boolean value can be used anywhere the search space is broken into 2 by single boundary. Here there is only one if statement checking for one condition. That's a job for boolean

Reply:
I am also doing the same thing as at the time of pushing it checks whether stack size limit is reached or not and if yes pop a fragment and then push the new one.So, where to use boolean field as such.

Reply:
To set a boolean value we need to first check the size (using if or getbackstackentrycount method) and now for next push we check this boolean value if its true pop element and do this.But what about if we press back then we need to update this boolean field accordingly.So, they both are one or the other way similar.

Reply:
Sir, in open-event-android app also the fragments are not added to backstack instead the fragment is replaced when pressed back button.

Reply:
I know they are replaceable, but the way I'm describing is clean design. Representing state by what it actually means. Think of it as a component which should abstract the working of navigation and only provide control for allowable things. Soon, I'll write a more descriptive solution

Reply:
I think Open Event Android has correct navigation flow

Reply:
No sir, in the open event android the code is not written using backstack, instead it is replaced. Please have a look at these screenshots.
![screenshot from 2018-03-22 14-58-12](https://user-images.githubusercontent.com/19494189/37780124-1aff6df6-2e14-11e8-8894-70f01cf30be9.png)
![screenshot from 2018-03-22 14-58-22](https://user-images.githubusercontent.com/19494189/37780130-1f19f69a-2e14-11e8-8b61-c42d5325763f.png)

Reply:
Sir, I'll do this issue please don't assign to others.

Reply:
I know it doesn't have backstack but its navigation is alright, we don't need backstack there